Create Deployment page (#5101)
mattermost/docs: source/about/deployments_and_editions.rst

Mattermost Deployments and Editions
===================================

> Mattermost Deployment Options

Reviewer: ```suggestion
Mattermost deployment options
```

------------------------------

Mattermost provides additional benefits in our commercial editions to support the hosting and growth needs of an organization. You can explore these commercial features through a `trial <https://mattermost.com/get-started/>`__ with Mattermost Self-Hosted or Mattermost Cloud deployment options.

> Mattermost Self-Hosted 

Reviewer: ```suggestion
Mattermost self-hosted 
```

Once you’ve downloaded and installed Mattermost within your preferred environment, you have the option to use it as-is, or you can access Mattermost's commercial features by starting a trial or purchasing a subscription. You can start a 30-day free Enterprise trial via **Main Menu > System Console > Edition and License > Start trial** or request a trial online at https://mattermost.com/trial/.

Mattermost Cloud
^^^^^^^^^^^^^^^^^

Mattermost Cloud delivers Mattermost as software-as-a-service in a highly secure, hosted environment designed for privacy-conscious enterprises and developer collaboration.

To get started with a 14-day free trial of Mattermost Cloud online, visit https://customers.mattermost.com/cloud/signup. Signing up requires you to create an account with a business email and specify a domain for the workspace. Once you sign up and create your workspace, a wizard will help you get started completing your profile, inviting team members, and integrating with your favorite developer tools.

> Mattermost Editions

Reviewer: ```suggestion
Mattermost editions
```

- `Advanced team permissions <https://docs.mattermost.com/onboard/advanced-permissions.html#team-override-schemes-e20>`__
- `Read-only announcement channels <https://docs.mattermost.com/manage/team-channel-members.html#channel-moderation-e20>`__
- `System-wide announcement banners <https://docs.mattermost.com/manage/announcement-banner.html>`__
- O365 integrations with `Microsoft Teams Calling <>`_ and `Microsoft Calendar <>`_
- 99.9% uptime SLA guarantee (Cloud only)
- `Next business day support via online ticketing system <https://mattermost.com/support/>`__

See a complete list of features `here <https://mattermost.com/pricing>`__.

Mattermost Enterprise 
^^^^^^^^^^^^^^^^^^^^^^

Mattermost Enterprise is an enterprise-grade collaboration system that supports and helps you scale your mission-critical enterprise workflows, meet strict enterprise security, compliance, and privacy requirements, as well as provide executive reporting, dashboards, and productivity metrics. 

Building on the features in Professional plan, it also offers:

- `Shared channels with other Mattermost instances <https://docs.mattermost.com/onboard/shared-channels.html>`__

> - `Configurable playbook permissions, analytics dashboards and channel exports <>`_

Reviewer: Does this need a docs link or will it link to the website? For a docs link, we don't have all the mentioned content on one page but this includes the permissions section: https://docs.mattermost.com/playbooks/setting-up-playbooks.html.

- `ID-only push notifications <https://docs.mattermost.com/configure/configuration-settings.html#push-notification-contents>`__
- `Global and custom retention policies for messages and files <https://docs.mattermost.com/comply/data-retention-policy.html>`__
- `Custom admin roles <https://docs.mattermost.com/onboard/system-admin-roles.html>`__
- `Granular audit logs with data export <https://docs.mattermost.com/comply/audit-log.html>`__
- `High availability support with multi-node database deployment <https://docs.mattermost.com/scale/high-availability-cluster.html>`__
- `Horizontal scaling through cluster-based deployment <https://docs.mattermost.com/scale/scaling-for-enterprise.html#cluster-based-deployment>`__
- `Elasticsearch support for highly efficient database searches in a cluster environment <https://docs.mattermost.com/scale/elasticsearch.html>`__
- `Advanced performance monitoring <https://docs.mattermost.com/scale/performance-monitoring.html>`__
- `Eligibility for Premier Support add-on <https://mattermost.com/support/>`__

See a complete list of features `here <https://mattermost.com/pricing/>`__. 

> 

Reviewer: ```suggestion
```

> Mattermost Legacy Enterprise Editions

Reviewer: ```suggestion
Mattermost legacy Enterprise Editions
```

- Advanced access control policy
- Next business day support via online ticketing system
- Scale to handle hundreds of users per team

Mattermost Enterprise Edition E20
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Mattermost Enterprise E20 was offered as a commercial enterprise-grade messaging system that scales from hundreds to tens of thousands of users. Enterprise Edition E20 authentication features are now offered in Mattermost Professional and High Availability and Compliance features are offered in Mattermost Enterprise. 

- Advanced SAML 2.0 authentication with Okta, OneLogin, and Active Directory Federation Services
- Active Directory/LDAP group synchronization
- OpenID Connect authentication for team creation, account creation, and user sign-in
- Compliance exports of message histories with oversight protection
- Custom retention policies for messages and files
- High Availability support with multi-node database deployment
- Horizontal scaling through cluster-based deployment
- Elasticsearch support for highly efficient database searches in a cluster environment
- Advanced performance monitoring
- Eligibility for Premier Support add-on

> 

Reviewer: ```suggestion
```

> Packaging Decisions

Reviewer: ```suggestion
Packaging decisions
```

> ----------------------------

Reviewer: ```suggestion
-----------------------
```

> As the platform matures and new features are added, they are evaluated to be included in the edition that best aligns with the organizational use cases outlined by the editions above. Multiple factors are considered in determining which tier to include a feature including mission-critical impact, relative value to a single team, cross-functional teams, and the enterprise, as well as security, compliance, and scalability.

Reviewer: ```suggestion
As the platform matures and new features are added, they're evaluated to be included in the edition that best aligns with the organizational use cases outlined by the editions above. Multiple factors are considered in determining which tier to include a feature including mission-critical impact, relative value to a single team, cross-functional teams, and the enterprise, as well as security, compliance, and scalability.
```

------------------------------

Mattermost provides additional benefits in our commercial editions to support the hosting and growth needs of an organization. You can explore these commercial features through a `trial <https://mattermost.com/get-started/>`__ with Mattermost Self-Hosted or Mattermost Cloud deployment options.  

Mattermost Self-Hosted 
^^^^^^^^^^^^^^^^^^^^^^

Mattermost Self-Hosted provides Mattermost within IT-controlled private environments in public clouds, including AWS and Azure, as well as on-premise in private clouds and virtual or physical servers.

Once you’ve downloaded and installed Mattermost within your preferred environment, you have the option to use it as-is, or you can access Mattermost's commercial features by starting a trial or purchasing a subscription. You can start a 30-day free Enterprise trial via **Main Menu > System Console > Edition and License > Start trial** or request a trial online at https://mattermost.com/trial/. 

Mattermost Cloud
^^^^^^^^^^^^^^^^^

Mattermost Cloud delivers Mattermost as software-as-a-service in a highly secure, hosted environment designed for privacy-conscious enterprises and developer collaboration. 

> To get started with a 14-day free trial of Mattermost Cloud online, visit https://mattermost.com/mattermost-cloud. Signing up requires you to create an account with a business email and specify a domain for the workspace. Once you sign up and create your workspace, a wizard will help you get started completing your profile, inviting team members, and integrating with your favorite developer tools. 

Reviewer 2: ```suggestion
To get started with a 14-day free trial of Mattermost Cloud online, visit https://customers.mattermost.com/cloud/signup. Signing up requires you to create an account with a business email and specify a domain for the workspace. Once you sign up and create your workspace, a wizard will help you get started completing your profile, inviting team members, and integrating with your favorite developer tools. 
```

- `Custom admin roles <https://docs.mattermost.com/onboard/system-admin-roles.html>`__
- `Granular audit logs with data export <https://docs.mattermost.com/comply/audit-log.html>`__
- `High availability support with multi-node database deployment <https://docs.mattermost.com/scale/high-availability-cluster.html>`__
- `Horizontal scaling through cluster-based deployment <https://docs.mattermost.com/scale/scaling-for-enterprise.html#cluster-based-deployment>`__
- `Elasticsearch support for highly efficient database searches in a cluster environment <https://docs.mattermost.com/scale/elasticsearch.html>`__
- `Advanced performance monitoring <https://docs.mattermost.com/scale/performance-monitoring.html>`__
- `Eligibility for Premier Support add-on <https://mattermost.com/support/>`__

See a complete list of features `here <https://mattermost.com/pricing/>`__. 

Mattermost Legacy Enterprise Editions
--------------------------------------

Mattermost introduced a new pricing and packaging structure on October 13, 2021. The packages listed below will reach end-of-life on October 31, 2023. 

> We are no longer selling these products to new customers. For existing customers, we highly recommend working with your Mattermost Account team to plan for a migration to our new packages, but we will honor existing pricing and features for renewals and expansions of E10/20 until September 30th, 2022. 

Reviewer 2: ```suggestion
We are no longer selling these products to new customers. For existing customers, we highly recommend working with your Mattermost Account team to plan for a migration to our new packages, but we will honor existing pricing and features for renewals and expansions of E10/20 until October 31, 2022. 
```
